packages/api/src/record/service.ts

```diff
@@ -32,6 +35,55 @@ export const getRecordById = async (requestQuery: {
   return records;
 };
 
+const validateCanPatchRecord = async (
+  recordId: string,
+  emailFromAuthToken: string
+): Promise<void> => {
+  const [record] = await getRecordById({
+    recordIds: [recordId],
+    accountEmail: emailFromAuthToken,
+  });
+  if (!record) {
+    throw new createError.NotFound("Record not found");
+  }
+  let canEdit = false;
+  const accountArchive = await accountService.getAccountArchive(
+    record.archiveId,
+    emailFromAuthToken
+  );
+  if (accountArchive && permission.checkCanEditAccountArchive(accountArchive)) {
+    canEdit = true;
+  }
+  if (!canEdit) {
+    const archiveMemberships =
+      await accountService.getCurrentAccountArchiveMemberships(
+        emailFromAuthToken
+      );
+    const editMemberships = await Promise.all(
+      archiveMemberships.map(async (archiveMembership) => {
+        if (permission.checkCanEditAccountArchive(archiveMembership)) {
+          const accessList = await folderAccess.getAccessByFolder(
```

Comment:
`access` entries are created for every shared item, including the descendants of shared folders, so you should check for an `access` entry for the record itself rather than its parent (the record could have been shared on its own, rather than as part of a folder).